DSharpPlus/Entities/Channel/Message/DiscordMessageSnapshot.cs

```diff
@@ -0,0 +1,15 @@
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities;
+
+/// <summary>
+/// Represents a Discord message snapshot.
+/// </summary>
+public class DiscordMessageSnapshot
+{
+    /// <summary>
+    /// Gets the message object for the message snapshot.
+    /// </summary>
+    [JsonProperty("message", NullValueHandling = NullValueHandling.Ignore)]
+    public DiscordMessage Message { get; set; }
```

Comment:
this needs to be its own type, the DiscordMessage object assumes it always has an ID, which this doesn't